pyproject.toml

```diff
@@ -1,7 +1,41 @@
+[project]
+name = "theHarvester"
+description = "theHarvester is a very simple, yet effective tool designed to be used in the early stages of a penetration test"
+readme = "README.md"
+authors = [
+    { name = "Christian Martorella", email = "cmartorella@edge-security.com" },
+]
+requires-python = ">=3.9"
+urls.Homepage = "https://github.com/laramies/theHarvester"
+classifiers = [
+    "Programming Language :: Python :: 3",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: 3.11",
+    "License :: OSI Approved :: GNU General Public License v2 (GPLv2)",
+    "Operating System :: OS Independent",
+]
+dynamic = ["dependencies", "optional-dependencies", "version"]
+
+[project.scripts]
+theHarvester = "theHarvester.theHarvester:main"
+restfulHarvest = "theHarvester.restfulHarvest:main"
+
+[tool.setuptools.dynamic]
+version = { attr = "theHarvester.lib.version.VERSION" }
+dependencies = { file = "requirements/base.txt" }
```

Comment:
Note I excluded this to keep my changes as minimal as possible, but we could also include the `requirements/dev.txt` as a `dev` extra:

```toml
optional-dependencies.dev = { file = "requirements/dev.txt" }
```

which would allow developers to install using the common `pip install -e '.[dev]'`. However, it doesn't support pip flags, like https://github.com/laramies/theHarvester/blob/f661afc2559eaba5daf15e36d7423672e1091a9b/requirements/dev.txt#L1
That line wouldn't be needed anymore since `pip` would already install the base requirements, but it would require a different workflow for devs as well as updating the wiki